src/JavisNB.jl

```diff
@@ -0,0 +1,187 @@
+#### Pluto Viewer
+#################
+
+"""
+    _pluto_viewer(video::Video, frames::Int, actions::Vector)
+
+Creates an interactive viewer in a Pluto Notebook by storing all the frames in-memory.
+
+# Returns
+
+`arr`: List of Javis frames
+"""
+function _pluto_viewer(video::Video, frames::Int, objects::Vector)
+    arr = collect(
+        Javis.get_javis_frame(video, objects, frame; layers = video.layers) for
+        frame in 1:frames
+    )
+    return arr
+end
+
+"""
+    PlutoViewer
+
+Wrapper to assist viewing rendered gifs as cell outputs of Pluto notebooks
+when `liveview = false`.
+"""
+struct PlutoViewer
+    filename::String
+end
+
+function Base.show(io::IO, ::MIME"image/png", v::PlutoViewer)
+    write(io, read(v.filename))
+end
+
+###################
+#### Jupyter Viewer
+################### 
+
+
+"""
+    _jupyter_viewer(video::Video, frames::Int, actions::Vector)
+Creates an interactive viewer in a Jupyter Notebook.
+"""
+function _jupyter_viewer(
+    video::Video,
+    frames::Int,
+    objects::Vector{AbstractObject},
+    framerate::Int,
+)
+    t = Interact.textbox(1:frames, value = 1, typ = Int)
+    f = Interact.slider(1:frames, label = "Frame", value = t)
+    obs = Interact.Observables.throttle(1 / framerate, f)
+    output = @map Javis.get_javis_frame(video, objects, &obs; layers = video.layers)
+    wdg = Widget(["f" => f, "t" => t], output = output)
+    @layout! wdg vbox(hbox(:f, :t), output)
+end
+
+
+##########
+#### embed
+##########
+
+"""
+    embed(pathname::String)
+
+Shows a gif within a notebook by calling the path to the gif.
+
+# Arguments
+    
+- `pathname::String`: the path to a gif to render in the notebook
+"""
+function embed(pathname::String)
+
+    if (isdefined(Main, :IJulia) && Main.IJulia.inited) || endswith(@__FILE__, ".ipynb")
+        display(MIME("text/html"), """<img src="$(pathname)">""")
+    elseif isdefined(Main, :PlutoRunner)
+        return PlutoViewer(pathname)
+    end
+
+    return pathname
+end
+
+function embed(vid::Video, frames::Int, objects::Vector{AbstractObject}, framerate::Int)
+    return _jupyter_viewer(vid, frames, objects, framerate)
+end
+
+function embed(vid::Video, frames::Int, objects::Vector{AbstractObject})
+    return _pluto_viewer(vid, frames, objects)
+end
+
+embed(::Nothing) = nothing
+
+
+"""
+    embed(
+        video::Video;
+        framerate=30,
+        pathname="javis_GIBBERISH.gif",
+        liveview=false,
+        streamconfig::Union{StreamConfig, Nothing} = nothing,
+        tempdirectory="",
+        ffmpeg_loglevel="panic",
+        rescale_factor=1.0,
+        postprocess_frame=identity,
+        postprocess_frames_flow=default_postprocess
+    )
+
+The core function of `JavisNB`.
+It allows one to show the output of `Javis.render` 
+from [Javis](https://juliaanimators.github.io/Javis.jl/stable/) directly within a notebook.
+Notebooks supported are: 
+
+- IJulia
+- Pluto
+
+To use it just replace `render` with `embed`, an example within `Pluto`:
```

Comment:
"To use it just replace `render` with `embed`, an example within `Pluto`:"

to

"To use `embed` just replace `Javis.render` with `embed`.
Here is an example one can use within a `Pluto` notebook:"